cmd/peer-rest-client.go

```diff
@@ -900,3 +902,58 @@ func newPeerRESTClient(peer *xnet.Host) *peerRESTClient {
 
 	return &peerRESTClient{host: peer, restClient: restClient}
 }
+
+// MonitorBandwidth - send http trace request to peer nodes
+func (client *peerRESTClient) MonitorBandwidth(ch chan interface{}, doneCh <-chan struct{}, buckets []string) {
```

Comment:
I still don't get why you aren't just sending a context here?

Reply:
the `doneCh` comes from the context created by the admin API, not sure how you would like to see the context being used in addition to the done channel.

Reply:
actually passing context is better because you can send that all way to the network call as well @kerneltime so the function would be 

```suggestion
func (client *peerRESTClient) MonitorBandwidth(ctx context.Context, ch chan interface{}, buckets []string) {
..
...
for {
    select {
    case <-ctx.Done():
    }
}

restClient.Call(ctx...)
```

Reply:
There are 2 contexts here, the context originating from the admin API and the global context, the global context and the context from the admin API is taken into consideration in the code path. In `doBandwidthMonitoring` 
```
	cancelCh := make(chan struct{})
	defer close(cancelCh)

	go func() {
		select {
		case <-doneCh:
		case <-cancelCh:
			// There was an error in the REST request.
		}
		cancel()
	}()
```

Reply:
you are not passing the context to network call @kerneltime that means it won't get canceled if the client disconnects.  We have context for this reason alone. 

the `newContext()` function here retains the connection context. The connection context always is based on GlobalContext which is a BaseContext and the context of the connection. 

Passing `doneCh` is like de-coupling which makes it complicated i.e functions don't get canceled explicitly if the client dies.

Reply:
If the intention is to monitor bandwidth across servers then you should use a more `pub/sub` model-like `TraceHandler`  rather than doing a continuous request every few seconds. 

Rather remote should send you the data.